src/demo/CMakeLists.txt

```
@@ -11,7 +11,18 @@ set(SRC
 add_executable(${TARGET} ${SRC})
 
 target_link_libraries(${TARGET} PRIVATE
-    sfml-system sfml-network sfml-graphics sfml-window ${PROJECT_NAME}_ecs Boost::thread pthread
+    ${PROJECT_NAME}_ecs Boost::thread pthread
 )
 
+if (CLIENT_COMPILATION_MODE)
+    target_link_libraries(${TARGET} PRIVATE
+        sfml-system sfml-network sfml-graphics sfml-window
```

Threads and Boost thread pools are also needed on the client
```suggestion
        sfml-system sfml-network sfml-graphics sfml-window Boost::thread pthread
```